src/libraries/System.Private.CoreLib/src/System/Runtime/InteropServices/MemoryMarshal.cs

```diff
@@ -211,27 +211,63 @@ public static ReadOnlySpan<byte> AsBytes<T>(ReadOnlySpan<T> span)
         }
 
         /// <summary>
-        /// Create a new span over a portion of a regular managed object. This can be useful
+        /// Creates a new span over a portion of a regular managed object. This can be useful
         /// if part of a managed object represents a "fixed array." This is dangerous because the
         /// <paramref name="length"/> is not checked.
         /// </summary>
         /// <param name="reference">A reference to data.</param>
         /// <param name="length">The number of <typeparamref name="T"/> elements the memory contains.</param>
-        /// <returns>The lifetime of the returned span will not be validated for safety by span-aware languages.</returns>
+        /// <returns>A span representing the specified reference and length.</returns>
+        /// <remarks>The lifetime of the returned span will not be validated for safety by span-aware languages.</remarks>
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
         public static Span<T> CreateSpan<T>(ref T reference, int length) => new Span<T>(ref reference, length);
 
         /// <summary>
-        /// Create a new read-only span over a portion of a regular managed object. This can be useful
+        /// Creates a new read-only span over a portion of a regular managed object. This can be useful
         /// if part of a managed object represents a "fixed array." This is dangerous because the
         /// <paramref name="length"/> is not checked.
         /// </summary>
         /// <param name="reference">A reference to data.</param>
         /// <param name="length">The number of <typeparamref name="T"/> elements the memory contains.</param>
-        /// <returns>The lifetime of the returned span will not be validated for safety by span-aware languages.</returns>
+        /// <returns>A read-only span representing the specified reference and length.</returns>
+        /// <remarks>The lifetime of the returned span will not be validated for safety by span-aware languages.</remarks>
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
         public static ReadOnlySpan<T> CreateReadOnlySpan<T>(ref T reference, int length) => new ReadOnlySpan<T>(ref reference, length);
 
+        /// <summary>Creates a new read-only span for a null-terminated string.</summary>
+        /// <param name="value">The pointer to the null-terminated string of characters.</param>
+        /// <returns>A read-only span representing the specified null-terminated string.</returns>
+        /// <remarks>The returned span does not include the null terminator.</remarks>
+        /// <exception cref="ArgumentNullException">The <paramref name="value"/> pointer is null.</exception>
+        /// <exception cref="ArgumentException">The string is longer than <see cref="int.MaxValue"/>.</exception>
+        [CLSCompliant(false)]
+        public static unsafe ReadOnlySpan<char> CreateReadOnlySpanFromNullTerminated(char* value)
+        {
+            if (value is null)
```

Comment:
Would it be more useful to return empty span instead of throwing? It is what string constructors do, e.g. `new string((char*)null)` returns empty string. It does not throw.

Reply:
The API proposal in https://github.com/dotnet/runtime/issues/40202 suggested that null should be non-throwing.

Reply:
Should we also return an empty/default span if length is 0 (and `value` is non-null)? The [API proposal](https://github.com/dotnet/runtime/issues/40202#issue-669850037) also special cases that, rather than returning `Span(non-null ptr, 0)`.

Reply:
I do not see a problem with returning `Span(non-null ptr, 0)`. Many other Span APIs do that.

Reply:
> Would it be more useful to return empty span instead of throwing?

I don't know that it's more useful, but consistency is good. I'll change it.